cmake/Modules/FindMKLDNN.cmake

```diff
@@ -85,13 +85,16 @@ IF(NOT MKLDNN_FOUND)
     SET(ONEDNN_BUILD_GRAPH ON CACHE BOOL "" FORCE)
   ENDIF(NOT APPLE AND NOT WIN32 AND NOT BUILD_LITE_INTERPRETER)
 
+  MESSAGE("-- Will build oneDNN UKERNEL")
+  SET(DNNL_EXPERIMENTAL_UKERNEL ON CACHE BOOL "" FORCE)
```

Comment:
How new is the ukernel feature? Perhaps you should check if dnnl_ukernel.hpp exists?

Reply:
This flag is necessary for building oneDNN with this feature, which was introduced in oneDNN 3.5. This change wouldn't have any functional side-effects if an older version of oneDNN would be used to build PyTorch, except that in that case, it'd incorrectly print in the PyTorch build log that this feature would be enabled.

Please advise if you meant moving this block under a conditional that checks for the presence of `dnnl_ukernel.hpp`.

Thanks!

Reply:
> This change wouldn't have any functional side-effects if an older version of oneDNN would be used to build PyTorch, except that it'd incorrectly print in the PyTorch build log that this feature would be enabled.

Wouldn't CPUBlas fail to build since it includes `dnnl_ukernel.hpp` whenever mkldnn is enabled? Maybe this is fine though, I don't know if we have a minimum required version.

Reply:
Added check for dnnl_ukernel.hpp.